### PR TITLE
[Formapp] migration to null safety

### PR DESCRIPTION
**Pull request by arteevraina**
Fixes: #771 

* Updated dependencies.
* Ran dart migrate.

cc @RedBrogdon @domesticmouse .

### CODE REVIEW

File: form_app/lib/src/validation.dart [hunk 1]
```diff
@@ -90,7 +90,7 @@ class _FormValidationDemoState extends State<FormValidationDemo> {
                 // A text field that validates that the text is a noun.
                 TextFormField(
                   validator: (value) {
-                    if (value.isEmpty) {
+                    if (value!.isEmpty) {
```
Comment (RedBrogdon) [inline form_app/lib/src/validation.dart:93]: How does this validator method handle the null case?

Would `value == null || value.isEmpty` be a better conditional here?

File: form_app/lib/src/validation.dart [hunk 2]
```diff
@@ -114,7 +114,7 @@ class _FormValidationDemoState extends State<FormValidationDemo> {
                 // checkbox.
                 FormField(
                   initialValue: false,
-                  validator: (value) {
+                  validator: (dynamic value) {
```
Comment (RedBrogdon) [inline form_app/lib/src/validation.dart:117]: Are you getting an error without this `dynamic`? I feel like it shouldn't be necessary, but I'm willing to be corrected. 😄

**Review by RedBrogdon — CHANGES_REQUESTED** [review]
Thanks for doing this migration!

Most of my comments are about converting fields to be non-nullable rather than just letting them allow nulls.

**Comment by arteevraina** [comment]
@RedBrogdon Updated my PR as per the above comments. But, there is an error coming which is possibly coming from [here](https://github.com/flutter/samples/blob/master/form_app/lib/src/http/mock_client.dart) on building `SignInHttpDemo`. I tried to look but could not figure out why it is coming. Do have any comments on this?

Error Details : 
```
The following _TypeError was thrown building Builder(dirty):
type 'Null' is not a subtype of type 'Future<Response>'

The relevant error-causing widget was:
  MaterialApp file:///home/arteev/samples/form_app/lib/main.dart:49:12
```

**Comment by domesticmouse** [comment]
> I tried to look but could not figure out why it is coming. Do have any comments on this?

I had a quick attempt at rebasing this PR up to the current version of master, and yes this error replicates. The generated stack trace looks something like this:

```console
══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
The following _TypeError was thrown building Builder(dirty):
type 'Null' is not a subtype of type 'Future<Response>'

The relevant error-causing widget was:
  MaterialApp file:///Users/brettmorgan/Documents/GitHub/samples/form_app/lib/main.dart:49:12

When the exception was thrown, this was the stack:
#0      MockClient.post (package:http/src/client.dart:62:20)
#1      new MockClient (package:form_app/src/http/mock_client.dart:12:10)
#2      httpClient (package:form_app/main.dart:15:32)
#3      httpClient (package:form_app/main.dart)
#4      demos.<anonymous closure> (package:form_app/main.dart:26:19)
#5      MaterialPageRoute.buildContent (package:flutter/src/material/page.dart:53:55)
#6      MaterialRouteTransitionMixin.buildPage (package:flutter/src/material/page.dart:106:27)
#7      _ModalScopeState.build.<anonymous closure>.<anonymous closure>
(package:flutter/src/widgets/routes.dart:843:53)
#8      Builder.build (package:flutter/src/widgets/basic.dart:7798:48)
#9      StatelessElement.build (package:flutter/src/widgets/framework.dart:4648:28)
#10     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4574:15)
#11     Element.rebuild (package:flutter/src/widgets/framework.dart:4267:5)
#12     ComponentElement._firstBuild (package:flutter/src/widgets/framework.dart:4553:5)
#13     ComponentElement.mount (package:flutter/src/widgets/framework.dart:4548:5)
...     Normal element mounting (15 frames)
#28     Element.inflateWidget (package:flutter/src/widgets/framework.dart:3611:14)
#29     MultiChildRenderObjectElement.inflateWidget
(package:flutter/src/widgets/framework.dart:6221:36)
#30     MultiChildRenderObjectElement.mount (package:flutter/src/widgets/framework.dart:6232:32)
...     Normal element mounting (170 frames)
#200    Element.inflateWidget (package:flutter/src/widgets/framework.dart:3611:14)
#201    MultiChildRenderObjectElement.inflateWidget
(package:flutter/src/widgets/framework.dart:6221:36)
#202    Element.updateChild (package:flutter/src/widgets/framework.dart:3363:18)
#203    RenderObjectElement.updateChildren (package:flutter/src/widgets/framework.dart:5654:32)
#204    MultiChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6243:17)
#205    Element.updateChild (package:flutter/src/widgets/framework.dart:3350:15)
#206    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4599:16)
#207    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4746:11)
#208    Element.rebuild (package:flutter/src/widgets/framework.dart:4267:5)
#209    StatefulElement.update (package:flutter/src/widgets/framework.dart:4778:5)
#210    Element.updateChild (package:flutter/src/widgets/framework.dart:3350:15)
#211    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4599:16)
#212    Element.rebuild (package:flutter/src/widgets/framework.dart:4267:5)
#213    ProxyElement.update (package:flutter/src/widgets/framework.dart:4922:5)
#214    Element.updateChild (package:flutter/src/widgets/framework.dart:3350:15)
#215    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4599:16)
#216    Element.rebuild (package:flutter/src/widgets/framework.dart:4267:5)
#217    ProxyElement.update (package:flutter/src/widgets/framework.dart:4922:5)
#218    _InheritedNotifierElement.update (package:flutter/src/widgets/inherited_notifier.dart:181:11)
#219    Element.updateChild (package:flutter/src/widgets/framework.dart:3350:15)
#220    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6090:14)
#221    Element.updateChild (package:flutter/src/widgets/framework.dart:3350:15)
#222    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4599:16)
#223    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4746:11)
#224    Element.rebuild (package:flutter/src/widgets/framework.dart:4267:5)
#225    StatefulElement.update (package:flutter/src/widgets/framework.dart:4778:5)
#226    Element.updateChild (package:flutter/src/widgets/framework.dart:3350:15)
#227    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6090:14)
#228    Element.updateChild (package:flutter/src/widgets/framework.dart:3350:15)
#229    SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6090:14)
#230    Element.updateChild (package:flutter/src/widgets/framework.dart:3350:15)
#231    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4599:16)
#232    Element.rebuild (package:flutter/src/widgets/framework.dart:4267:5)
#233    ProxyElement.update (package:flutter/src/widgets/framework.dart:4922:5)
#234    Element.updateChild (package:flutter/src/widgets/framework.dart:3350:15)
#235    ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4599:16)
#236    StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4746:11)
#237    Element.rebuild (package:flutter/src/widgets/framework.dart:4267:5)
#238    BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2582:33)
#239    WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:875:21)
#240    RendererBinding._handlePersistentFrameCallback
(package:flutter/src/rendering/binding.dart:328:5)
#241    SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1144:15)
#242    SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1082:9)
#243    SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:998:5)
#247    _invoke (dart:ui/hooks.dart:163:10)
#248    PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:259:5)
#249    _drawFrame (dart:ui/hooks.dart:126:31)
(elided 3 frames from dart:async)

════════════════════════════════════════════════════════════════════════════════════════════════════
```

At a guess something is sideways in the configuration of the `MockClient`? 

How I'd progress this is I would revert to the point where things were working, then single step through adding the various changes between the working state and the current broken state, to figure out which of the changes broke the world. If you have small enough commits, you may even be able to get some assistance from `git bisect`.

**Comment by RedBrogdon** [comment]
CC: @johnpryan. 

The version bump for mockito is the likely culprit. In migrating it to null safety, the team had to make some pretty serious breaking changes, and using mockito now involves generated files and some other API changes which I don't believe this PR is addressing.

I can see a few ways out of this:

* Update the app to use the new version of mockito correctly.
* Remove mockito entirely and create a new MockClient without it that behaves the same.
* Remove the use of HttpClient entirely and replace it with an object that stands in for a network service wrapper (AuthService or something) and exposes a single method that accepts a username and password and returns a Future<bool> or Future<YourEnumHere>.

If it were me, I'd go with the last suggestion, since it's dead simple and doesn't get in the way of the goal of this sample (teach people how to use forms). This was @johnpryan's app, though, and I don't want to speak for him.

File: form_app/lib/src/validation.dart (hunk 1 above)
Comment (domesticmouse) [reply form_app/lib/src/validation.dart]: How about `value?.isEmpty ?? false`?

**Comment by Hixie** [comment]
@arteevraina Are you interested in still working on this?

**Comment by arteevraina** [comment]
@Hixie Would like to close this PR. This issue can be assigned to someone else.

**Pull request closed without merging**